src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,37 +75,231 @@ const NotebookLauncher = _.flow(
+      ['PREVIEW (READ-ONLY)']),
+    !readOnlyAccess && Utils.cond(
+      [
+        !mode, () => h(Fragment, [
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', paddingTop: '1rem', paddingBottom: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
+            onClick: () => locked ? setFileInUseOpen(true) : chooseMode('edit')
+          }, [icon('lock', { style: { paddingRight: '3px' } }), locked ? 'EDIT (IN USE)' : 'EDIT']),
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
+            onClick: () => {
+              hidePlaygroundModal === 'true' ? chooseMode('playground') : setPlaygroundModalOpen(true)
+            }
+          }, [icon('chalkboard', { style: { paddingRight: '3px' } }), 'PLAYGROUND MODE']),
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', backgroundColor: colors.dark(0.1), height: '100%' }
+          }, [
+            h(PopupTrigger, {
+              closeOnClick: true,
+              content: h(Fragment, [
+                h(MenuButton, { onClick: () => setCopyingNotebook(true) }, ['Make a Copy']),
+                h(MenuButton, { onClick: () => setExportingNotebook(true) }, ['Copy to another workspace']),
+                h(MenuButton, {
+                  onClick: async () => {
+                    try {
+                      await clipboard.writeText(`${window.location.host}/${notebookLink}`)
+                      notify('success', 'Successfully copied URL to clipboard', { timeout: 3000 })
+                    } catch (error) {
+                      reportError('Error copying to clipboard', error)
+                    }
+                  }
+                }, ['Copy URL to clipboard'])
+              ]),
+              side: 'bottom'
+            }, [
+              h(Clickable, { 'aria-label': 'Notebook options' }, [icon('ellipsis-v', {})])
+            ])
+          ])
+        ])
+      ],
+      [
+        clusterStatus === 'Creating', () => h(StatusMessage, { showSpinner: true }, [
+          'Creating notebook runtime environment, this will take 5-10 minutes. You can navigate away and return when it’s ready.'
+        ])
+      ],
+      [
+        clusterStatus === 'Starting', () => h(StatusMessage, { showSpinner: true }, [
+          'Starting notebook runtime environment, this may take up to 2 minutes.'
+        ])
+      ],
+      [
+        clusterStatus === 'Stopping', () => h(StatusMessage, { showSpinner: true }, [
+          'Notebook runtime environment is stopping. You can restart it after it finishes.'
+        ])
+      ],
+      [clusterStatus === 'Error', () => h(StatusMessage, ['Notebook runtime error.'])]
```

Comment:
Without cases for `null` or `'Stopped'`, it's possible to get to a dead end where there's no messaging, and the only clickable elements take you away from the page. We should probably provide some messaging similar to what the previous code did, and also maybe provide a way to 'cancel' out of edit mode and return to preview mode.